spring-cloud-alibaba-starters/spring-cloud-alibaba-governance/governance-core/spring-cloud-starter-alibaba-governance-labelrouting/label-routing-data/src/main/java/com/alibaba/cloud/router/data/repository/RouteDataRepository.java

```diff
@@ -16,113 +16,247 @@
-		LabelRouteData putLabelRouteData = null;
-
-		for (UntiedRouteDataStructure routerData : routerDataList) {
-			putLabelRouteData = routeCache.put(routerData.getTargetService(),
-					routerData.getLabelRouteData());
-			if (putLabelRouteData == null) {
-				log.info("Label route rule:" + routerData
-						+ "had been add to router cache");
-			}
-		}
-	}
-
-	public LabelRouteData getRouteData(String targetService) {
-		// double check.
+	private void updateData(String targetService) {
 		while (routeDataChanged) {
-			updateRouteData();
-
+			// Update label rule data.
+			int updateIndex = updateRouteData();
+			// Double check.
 			if (routeDataChanged) {
 				int matchIndex = 0;
+				// Find targetService index in list.
 				for (UntiedRouteDataStructure routeData : routeDataList) {
 					if (targetService.equals(routeData.getTargetService())) {
 						break;
 					}
 					matchIndex++;
 				}
-				if (matchIndex <= updateIndex.get()) {
-					return routeCache.get(targetService);
+				// If match index has update.
+				if (matchIndex <= updateIndex) {
+					return;
+				}
+			}
+		}
+	}
+
+	private void nonNullCheck(UntiedRouteDataStructure untiedRouteDataStructure) {
+		String targetService = untiedRouteDataStructure.getTargetService();
+		if (targetService == null) {
+			LOG.error("Lose target Service name.");
+		}
+		final LabelRouteData labelRouteData = untiedRouteDataStructure
+				.getLabelRouteData();
+		final List<MatchService> matchServiceList = labelRouteData.getMatchRouteList();
+		for (MatchService matchService : matchServiceList) {
+			final List<RouteRule> ruleList = matchService.getRuleList();
+			String version = matchService.getVersion();
+			Integer weight = matchService.getWeight();
+			if (CollectionUtils.isEmpty(ruleList)) {
+				LOG.error("Rule is empty in version = {} ", version);
+			}
+			if (version == null) {
+				LOG.error("Target service = {} lose version,please check it. ",
+						targetService);
+			}
+			if (weight == null) {
+				weight = DEFAULT_WEIGHT;
+			}
+			if (weight < MIN_WEIGHT || weight > SUM_WEIGHT) {
+				LOG.error(
+						"The weight of provider = {} version = {} had set error,please check it. ",
+						targetService, version);
+			}
+		}
+	}
+
+	private void putRouteData(final List<UntiedRouteDataStructure> routerDataList) {
+		for (UntiedRouteDataStructure routerData : routerDataList) {
+			buildHashIndex(routerData);
+			originalRouteData.put(routerData.getTargetService(),
+					routerData.getLabelRouteData());
+		}
+	}
+
+	private void buildHashIndex(final UntiedRouteDataStructure routerData) {
+		final List<MatchService> matchRouteList = routerData.getLabelRouteData()
+				.getMatchRouteList();
+		HashMap<String, List<MatchService>> singleRuleMap = new HashMap<>();
+
+		for (MatchService matchService : matchRouteList) {
+			List<RouteRule> ruleList = matchService.getRuleList();
+
+			// Take out the path label separately, because there is no key for hash index.
+			if (ruleList.size() == 1
```

Comment:
I don't understand here why need to do like your code when the rule is single path rule?

Reply:
path has no key